prow/config/config.go

```diff
@@ -264,6 +337,76 @@ func Load(path string) (*Config, error) {
 	return nc, nil
 }
 
+// mergeConfig merges two prow configs together
+// It will try to merge:
+//	- Presubmits
+//	- Postsubmits
+// 	- Periodics
+//	- PodPresets
+// It will reject and return an error if
+//	- multiple Prow config is defined
+func (c *Config) mergeConfig(nc *Config) error {
+	c.Presets = append(c.Presets, nc.Presets...)
+
+	c.Periodics = append(c.Periodics, nc.Periodics...)
+
+	for repo, jobs := range nc.Presubmits {
+		if _, ok := c.Presubmits[repo]; ok {
+			c.Presubmits[repo] = append(c.Presubmits[repo], jobs...)
+		}
+	}
+
+	for repo, jobs := range nc.Postsubmits {
+		if _, ok := c.Postsubmits[repo]; ok {
+			c.Postsubmits[repo] = append(c.Postsubmits[repo], jobs...)
+		}
+	}
+
+	// validate no duplicated presets
+	validLabels := map[string]string{}
+	for _, preset := range c.Presets {
+		for label, val := range preset.Labels {
+			if _, ok := validLabels[label]; ok {
+				return fmt.Errorf("Duplicated preset label : %s", label)
+			} else {
+				validLabels[label] = val
+			}
+		}
+	}
+
+	// validate no duplicated periodics
+	validPeriodics := map[string]bool{}
+	for _, p := range c.AllPeriodics() {
+		if _, ok := validPeriodics[p.Name]; ok {
+			return fmt.Errorf("Duplicated periodic job : %s", p.Name)
+		} else {
+			validPeriodics[p.Name] = true
+		}
+	}
+
+	// validate no duplicated presubmits
+	validPresubmits := map[string]bool{}
+	for _, p := range c.AllPresubmits(nil) {
+		if _, ok := validPresubmits[p.Name]; ok {
+			return fmt.Errorf("Duplicated presubmit job : %s", p.Name)
```

Comment:
This means that we won't allow multiple jobs with the same name in different job config files, but we will allow multiple jobs with the same name within a job config.
I think this is reasonable until we can switch to unique job names, but we should probably document this.

Reply:
I remember we have a presubmit that tests the same behavior, but I cannot find the code anymore :-(